generator/src/main/java/com/algorand/sdkutils/listeners/Publisher.java

```diff
@@ -0,0 +1,81 @@
+package com.algorand.sdkutils.listeners;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+import com.algorand.sdkutils.listeners.Publisher.Events;
+import com.algorand.sdkutils.utils.StructDef;
+import com.algorand.sdkutils.utils.TypeDef;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class Publisher {
```

Comment:
I'm a fan of guava's EventBus for a loosely coupled publish/subscribe library: https://github.com/google/guava/wiki/EventBusExplained

No need to change over since you've already written it